synapseclient/models/curation.py

```diff
@@ -932,3 +1165,115 @@ async def export_to_record_set_async(
+            grid.etag = session_dict.get("etag", None)
+            grid.modified_on = session_dict.get("modifiedOn", None)
+            grid.last_replica_id_client = session_dict.get("lastReplicaIdClient", None)
+            grid.last_replica_id_service = session_dict.get(
+                "lastReplicaIdService", None
+            )
+            grid.grid_json_schema_id = session_dict.get("gridJsonSchema$Id", None)
+            grid.source_entity_id = session_dict.get("sourceEntityId", None)
+            yield grid
+
+    @classmethod
+    def list(
+        cls,
+        source_id: Optional[str] = None,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> Generator["Grid", None, None]:
+        """
+        Generator to get a list of active grid sessions for the user.
+
+        Arguments:
+            source_id: Optional. When provided, only sessions with this synId will be returned.
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Yields:
+            Grid objects representing active grid sessions.
+        """
+        return wrap_async_generator_to_sync_generator(
+            async_gen_func=cls.list_async,
+            source_id=source_id,
+            synapse_client=synapse_client,
+        )
+
+    async def delete_async(self, *, synapse_client: Optional[Synapse] = None) -> None:
+        """
+        Delete the grid session.
+
+        Note: Only the user that created a grid session may delete it.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError: If session_id is not provided.
+        """
+        if not self.session_id:
+            raise ValueError("session_id is required to delete a GridSession")
+
+        trace.get_current_span().set_attributes(
+            {
+                "synapse.session_id": self.session_id or "",
+            }
+        )
+
+        await delete_grid_session(
+            session_id=self.session_id, synapse_client=synapse_client
+        )
+
+    def delete(self, *, synapse_client: Optional[Synapse] = None) -> None:
+        """
+        Delete the grid session.
+
+        Note: Only the user that created a grid session may delete it.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError: If session_id is not provided.
+        """
+        return None
```

Comment:
I'm assuming this will be filled out?

Reply:
I guess I'm confused. I thought the way it worked was that you define the async method in the async class ( `Grid.delete_async`), then you define the sync method in the Protocol class (`GridSynchronousProtocol.delete`). But you don't define the sync method in the async class (`Grid.delete`)

Reply:
Both work just fine.

But I realize that this should get moved over to GridSynchronousProtocol.

The reason it works is that adding classes to a parent class in Python works just the same as if the method were added directly to the parent class like this.